gateway/ln-gateway/src/client.rs

```diff
@@ -48,21 +47,31 @@ impl StandardGatewayClientBuilder {
         tg: TaskGroup,
         old_client: Option<fedimint_client::Client>,
     ) -> Result<fedimint_client::Client> {
-        let federation_id = config.config.federation_id;
+        let FederationConfig {
+            connection,
+            mint_channel_id,
+            timelock_delta,
+            fees,
+        } = config;
+        let federation_id = connection.id;
+
+        // TODO: remove after <https://github.com/fedimint/fedimint/issues/2506>
```

Comment:
What is the plan after this issue is resolved?

Reply:
gateway config builder does not have to download configs. Instead, the client builder will take over that responsibility, plus persisting downloaded configs